packages/eas-cli/src/__tests__/commands/authCommand-test.ts

```diff
@@ -0,0 +1,50 @@
+import { flushAsync, initAsync } from '../../analytics';
+import AuthorizedCommand from '../../commands/abstract/authorizedCommand';
+import { jester as mockJester } from '../../credentials/__tests__/fixtures-constants';
+import { ensureLoggedInAsync } from '../../user/actions';
+import TestAuthrizedCommand from './TestAuthrizedCommand';
+
+describe(AuthorizedCommand.name, () => {
+  beforeAll(() => {
+    jest.mock('../../user/actions', () => ({ ensureLoggedInAsync: jest.fn(() => mockJester) }));
+    jest.mock('../../analytics', () => ({ initAsync: jest.fn(), flushAsync: jest.fn() }));
+  });
+
+  afterEach(() => {
+    (ensureLoggedInAsync as jest.Mock).mockClear();
+    (initAsync as jest.Mock).mockClear();
+    (flushAsync as jest.Mock).mockClear();
+  });
+
+  describe('without exceptions', () => {
+    it('ensures the user is logged in', async () => {
+      await TestAuthrizedCommand.run();
+
+      expect(ensureLoggedInAsync).toHaveReturnedWith(mockJester);
+    });
+
+    it('initializes analytics', async () => {
+      await TestAuthrizedCommand.run();
+
+      expect(initAsync).toHaveBeenCalled();
+    });
+
+    it('flushes analytics', async () => {
+      await TestAuthrizedCommand.run();
+
+      expect(flushAsync).toHaveBeenCalled();
+    });
+  });
+
+  describe('after exceptions', () => {
```

Comment:
It's pretty difficult to test for exceptions without oclif's `fancy-test`. I was finding buggy behavior where jest said an exception was not thrown when it was, etc.

There is [experimental support](https://github.com/oclif/fancy-test#why) for running `fancy-test` with jest but I didn't want to bring that baggage into the pr.